zerver/lib/subscription_info.py

```diff
@@ -0,0 +1,287 @@
+import datetime
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+from typing_extensions import NotRequired, TypedDict
+
+from zerver.lib.email_mirror_helpers import encode_email_address_helper
+from zerver.lib.stream_color import STREAM_ASSIGNMENT_COLORS
+from zerver.lib.stream_traffic import get_average_weekly_stream_traffic
+from zerver.lib.streams import get_web_public_streams_queryset
+from zerver.lib.timestamp import datetime_to_timestamp
+from zerver.models import Realm, Stream, Subscription, UserProfile
+
+
+class RawStreamDict(TypedDict):
+    date_created: datetime.datetime
+    description: str
+    email_token: str
+    first_message_id: Optional[int]
+    history_public_to_subscribers: Optional[bool]
+    id: int
+    invite_only: Optional[bool]
```

Comment:
I think it's a bug that `invite_only` is flagged as `null=True`; it doesn't make sense logically and I've confirmed this configuration dates to 7d05450411dcb7e8fd3f55f3db6c2a2d1f69a97c.

>    invite_only: Optional[bool] = models.BooleanField(null=True, default=False)                          

The same applies here:
>    is_muted: Optional[bool] = models.BooleanField(null=True, default=False)                             

We should fix these, but it requires a database migration (which may need to set any existing values to the default value of `false`)